### PR TITLE
add cpp solution for 809

### PR DESCRIPTION
**Pull request by jjliewie**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which includes the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

File: solutions/usaco-809.mdx [hunk 1]
```diff
@@ -131,4 +131,53 @@ else:
...
 
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+int x[111], y[111], z[111];
+
+int main(){
```
Comment (ryanchou-dev) [inline solutions/usaco-809.mdx:142]:
```suggestion
int main() {
```

File: solutions/usaco-809.mdx [hunk 2]
```diff
@@ -131,4 +131,53 @@ else:
 
 </PySection>
 
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+int x[111], y[111], z[111];
```
Comment (ryanchou-dev) [inline solutions/usaco-809.mdx:140]: use a named constant instead

File: solutions/usaco-809.mdx [hunk 3]
```diff
@@ -131,4 +131,53 @@ else:
...
+
+int main(){
+
+	freopen("taming.in", "r", stdin);
+	freopen("taming.out", "w", stdout);
+
+	cin.tie(0)->sync_with_stdio(0);
```
Comment (ryanchou-dev) [inline solutions/usaco-809.mdx:147]:
```suggestion
	cin.tie(0) -> sync_with_stdio(0);
```

File: solutions/usaco-809.mdx [hunk 4]
```diff
@@ -131,4 +131,53 @@ else:
...
+
+	int n; cin >> n;
+	for (int i = 1; i <= n; i++) {
+		cin >> x[i];
+	}
+
+	y[1] = 1;
+	for (int i = 1; i <= n; i++) {
+		if(x[i] >= 0) {
```
Comment (ryanchou-dev) [inline solutions/usaco-809.mdx:156]:
```suggestion
	int n;
	cin >> n;
	for (int i = 1; i <= n; i++) {
		cin >> x[i];
	}

	y[1] = 1;
	for (int i = 1; i <= n; i++) {
		if (x[i] >= 0) {
```

**Review by ryanchou-dev — COMMENTED** [review]
you account for 72% of the prs... 🎄

File: solutions/usaco-809.mdx (hunk 2 above)
Comment (jjliewie) [reply solutions/usaco-809.mdx:140]: okay i'll change it

**Comment by jjliewie** [comment]
> you account for 72% of the prs... 🎄

nice

File: solutions/usaco-809.mdx [hunk 5]
```diff
@@ -131,4 +131,54 @@ else:
...
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+int x[111], y[111], z[111];
+
+int main() {
+
```
Comment (ryanchou-dev) [inline solutions/usaco-809.mdx:143]:
```suggestion
```

**Review by ryanchou-dev — CHANGES_REQUESTED** [review]
bump ^^

File: solutions/usaco-809.mdx (hunk 2 above)
Comment (ryanchou-dev) [reply solutions/usaco-809.mdx:140]: bump

**Review by envyaims — CHANGES_REQUESTED** [review]
uhhh i don't know if I'm missing something but what does any of these variables mean? Comments would be heavily appreciated. Your code also looks nothing like the code from other languages nor the official editorial so I have no idea what's going on.

**Review by ryanchou-dev — COMMENTED** [review]
yep ^^

adding on to envy's suggestions, a good rule of thumb is that if you don't have any comments/time complexity or any extra information, it's best to keep it as a user solution

**Comment by jjliewie** [comment]
> uhhh i don't know if I'm missing something but what does any of these variables mean? Comments would be heavily appreciated. Your code also looks nothing like the code from other languages nor the official editorial so I have no idea what's going on.

okay ill add some comments.. python is slow so i took a different approach.

File: solutions/usaco-809.mdx [hunk 6]
```diff
@@ -131,4 +131,53 @@ else:
...
+#include <bits/stdc++.h>
+using namespace std;
+
+int x[111], y[111], z[111];
+
+int main() {
+	freopen("taming.in", "r", stdin);
+	freopen("taming.out", "w", stdout);
+
+	cin.tie(0) -> sync_with_stdio(0);
```
Comment (SansPapyrus683) [inline solutions/usaco-809.mdx:146]: remove if runs fine without

File: solutions/usaco-809.mdx [hunk 7]
```diff
@@ -131,4 +131,53 @@ else:
...
+	} 
+
+	int first = 0, second = 0;
+
+	for (int i = 1; i <= n; i++) {
+		first += y[i];
+		second += !z[i];
+	} 
+	
+	cout << first << " " << second;
```
Comment (SansPapyrus683) [inline solutions/usaco-809.mdx:177]: add endl

File: solutions/usaco-809.mdx [hunk 8]
```diff
@@ -131,4 +131,53 @@ else:
...
+		}
+	}
+
+	for (int i = 1; i <= n; i++) {
+		if (y[i] && z[i]) {
+			return cout << -1, 0;
+		}
+	} 
+
+	int first = 0, second = 0;
```
Comment (SansPapyrus683) [inline solutions/usaco-809.mdx:170]: declarations on separate lines

File: solutions/usaco-809.mdx [hunk 9]
```diff
@@ -131,4 +131,53 @@ else:
...
+		if (x[i] >= 0) {
+			for (int j = i; j > i - x[i]; j--) {
+				z[j] = 1;
+			}
+			y[i - x[i]] = 1;
+		}
+	}
+
+	for (int i = 1; i <= n; i++) {
+		if (y[i] && z[i]) {
```
Comment (SansPapyrus683) [inline solutions/usaco-809.mdx:165]: explicitly say `y[i] != 0` and stuff like that

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

**Review by ryanchou-dev — COMMENTED** [review]
@jjliewie please address changes & conflicts 😭

File: solutions/usaco-809.mdx (hunk 6 above)
Comment (jjliewie) [reply solutions/usaco-809.mdx:146]: okay ill remove

**Comment by jjliewie** [comment]
> @jjliewie please address changes & conflicts 😭

addressed & resolved

File: solutions/usaco-809.mdx [hunk 10]
```diff
@@ -215,4 +215,54 @@ else:
...
 
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+// 111 is maximum size needed (with leeway for edge cases)
+
+int x[111], y[111], z[111];
```
Comment (SansPapyrus683) [inline solutions/usaco-809.mdx:226]: put in constant please

Comment (jjliewie) [reply solutions/usaco-809.mdx:226]: done

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/usaco-809.mdx [hunk 11]
```diff
@@ -215,4 +215,56 @@ else:
 
 </PySection>
 
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+// 111 is maximum size needed (with leeway for edge cases)
```
[the remark this answers is not in the document]
Comment (jjliewie) [reply solutions/usaco-809.mdx:224]: changing comment it is the maximum space needed

**Review by ryanchou-dev — APPROVED** [review]
\>W<

File: solutions/usaco-809.mdx [hunk 12]
```diff
@@ -215,4 +215,56 @@ else:
...
 </PySection>
 
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+// 111 is maximum size needed
+const int MAX_SIZE = 111;
```
[the remark this answers is not in the document]
Comment (SansPapyrus683) [reply solutions/usaco-809.mdx:225]: unresolved

**Comment by SansPapyrus683** [comment]
still some problems, mind if i just change it myself?

**Comment by SansPapyrus683** [comment]
actually i'm just gonna let this die bc of the conflicts

**Pull request closed without merging**